code/iaas/allocator/src/main/java/io/cattle/platform/allocator/dao/impl/AllocatorDaoImpl.java

```diff
@@ -170,6 +176,39 @@ protected void modifyCompute(long hostId, Instance instance, boolean add) {
         objectManager.persist(host);
     }
 
+    protected void modifyDisk(long hostId, Instance instance, boolean add) {
+        CacheManager cm = CacheManager.getCacheManagerInstance(this.objectManager);
+        InstanceInfo instanceInfo = cm.getInstanceInfoForHost(hostId, instance.getId());
+
+        for (Entry<String, InstanceDiskReserveInfo> entry : instanceInfo.getAllReservedDisksInfo()) {
```

Comment:
This comment applies to everything within this for loop. We do multiple mutations to the cache in this loop. Since this group of mutations as a whole is not atomic, we run the risk of data corruption in our cache.